src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1842,30 +1950,32 @@ int fmi3_xml_handle_StringVariable(fmi3_xml_parser_context_t *context, const cha
         if(!variable->type) return -1;
     }
     else {
-        /* set start value */
-
-        /* We must wait until after parsing dimensions, because we can't otherwise know
-         * if it's a 1x1 array or a scalar variable just by reading the start value. */
-        if (fmi3_xml_get_variable_has_start(variable)) {
-            if (fmi3_xml_variable_is_array(variable)) {
-                fmi3_xml_string_variable_start_t* start_obj =
-                        (fmi3_xml_string_variable_start_t*)fmi3_xml_alloc_variable_type_start(
-                                td, variable->type, sizeof(fmi3_xml_string_variable_start_t));
-                if (!start_obj) {
-                    fmi3_xml_parse_fatal(context, "Could not allocate memory");
+        /* Cannot use *has_start for strings */
+        size_t nStart = jm_vector_get_size(jm_voidp)(&context->currentStartVariableValues);
+        if (nStart > 0) {
+            fmi3_xml_string_variable_start_t* startObj =
+                    (fmi3_xml_string_variable_start_t*)fmi3_xml_alloc_variable_type_start(
+                            td, variable->type, sizeof(fmi3_xml_string_variable_start_t));
+            if (!startObj) {
+                fmi3_xml_parse_fatal(context, "Could not allocate memory");
+                return -1;
+            }
+            jm_vector_init(jm_voidp)(&startObj->stringStartValues, 0, context->callbacks);
+            // Passing ownership to stringStartValues
+            size_t nCopied = jm_vector_copy(jm_voidp)(&startObj->stringStartValues, &context->currentStartVariableValues);
+            variable->type = &startObj->super;
+            // Resize the vector to 0 since we are now done with the previous values.
+            jm_vector_resize(jm_voidp)(&context->currentStartVariableValues, 0);
+            if ((fmi3_xml_variable_is_array(variable))) {
+                if (nStart != nCopied) {
+                    fmi3_xml_parse_fatal(context, "Could not retrieve string start values");
                     return -1;
                 }
-                jm_vector_init(jm_voidp)(&start_obj->arrayStartValues, 0, context->callbacks);
-                size_t n = jm_vector_copy(jm_voidp)(&start_obj->arrayStartValues, &context->variableStartValues);
-                variable->type = &start_obj->super;
-                size_t s = jm_vector_get_size(jm_voidp)(&context->variableStartValues);
-                if (s != n) {
+            } else {
+                if (nStart != 1) {
                     fmi3_xml_parse_fatal(context, "Could not retrieve string start values");
                     return -1;
                 }
```

Comment:
What is the proposed change here? To me it seems reasonable to check that it `if (nStart != 1) ` because we know it is not an array. If `nStart != 1` evaluates to `true` it seems reasonable to break immediately.